tests/utils.py

```diff
@@ -487,8 +492,17 @@ def write(self, spans=None):
         if spans:
             traces = [spans]
             self.json_encoder.encode_traces(traces)
-            self.msgpack_encoder.put(spans)
-            self.msgpack_encoder.encode()
+            if self._trace_flush_enabled:
+                AgentWriter.write(self, spans=spans)
+            else:
+                self.msgpack_encoder.put(spans)
```

Comment:
Why did we encode them originally? To make sure performance was as expected, or do we sometimes look at them later, once encoded?

I'm only worried about this because I don't want to accidentally change a performance metric just because we're now writing rather than encoding sometimes. It's fine that we change, it'd just be good to know if we expect to see any performance change.

Reply:
Those encoded traces are occasionally looked at by some tests like tracer / writer tests. I noticed messing with that encoder caused failures from other tests using it, that's why I went with just writing using an AgentWriter instead.